subprojects/gradle/impact-shared/src/main/java/com/avito/impact/configuration/SimpleConfiguration.kt

```diff
@@ -91,25 +92,16 @@ abstract class SimpleConfiguration(val module: InternalModule) : Equality {
             .toSet()
     }
 
+    open val dependencies: Set<ImplementationConfiguration> by lazy {
+        module.project.dependenciesOnProjects(type)
+            .map {
+                it.dependencyProject
+                    .internalModule
+                    .implementationConfiguration
+            }
+            .toSet()
+    }
+
     protected abstract fun containsBytecode(bytecodeDirectory: File): Boolean
     protected abstract fun containsSources(sourceSet: AndroidSourceSet): Boolean
 }
-
-internal fun SimpleConfiguration.dependencies(filter: (Configuration) -> Boolean): Set<ImplementationConfiguration> {
-    return module.project.configurations
-        .filter(filter)
-        .flatMap { configuration ->
-            configuration
-                .dependencies
-                .withType(DefaultProjectDependency::class.java)
-        }
-        .toSet()
-        .map {
-            it.dependencyProject
-                .internalModule
-                .implementationConfiguration
-        }
-        .toMutableSet().apply {
```

Comment:
@eugene-krivobokov I don't get what it is. I expect the existing unit tests for that scenario.
Could we discuss and write them?

Reply:
Ok, I find that test